src/components/ProductItem/index.js

```diff
@@ -0,0 +1,59 @@
+import PropTypes from "prop-types";
+import { useCart } from "../../hooks";
+import { Card, Img, Price } from "./styled";
+import { Button, StyledLink } from "../../styled";
+import { formatter } from "../../utils";
+
+function ProductItem(props) {
+  const { id, data } = props.product;
+  const { format } = formatter;
+  const { dispatch, add, products } = useCart();
+
+  const currQty = products.find((p) => p.id === id)?.quantity || 0;
+  const isDisabled = currQty === data.stock;
+
+  const handleClick = () => {
+    if (!isDisabled) {
+      dispatch(add({ product: props.product, quantity: 1 }));
+    }
+  };
+
+  return (
+    <Card>
+      <StyledLink to={`${props.path}/${id}`}>
+        <Img src={data.mainimage.url} alt={data.mainimage.alt} />
+        <h4>{data.name}</h4>
+        <Price>{format(data.price)}</Price>
+        <p>{data.category.slug}</p>
+      </StyledLink>
+
+      <Button onClick={handleClick} disabled={isDisabled}>
+        {isDisabled ? "Out of stock" : "Add to cart"}
```

Comment:
Checkout the other message about the screen readers here as well. Also this looks like the same button, is it worth isolating this as a new component that you could use for both cases?

Reply:
I created a component to use in both places.